src/sbml/test/TestListOf.c

```diff
@@ -215,6 +215,36 @@ START_TEST (test_ListOf_append)
 }
 END_TEST
 
+START_TEST(test_ListOf_sort)
+{
+    ListOf_t* lo = (ListOf_t*)ListOf_create(2, 4);
+
+    Species_t* sp = Species_create(2, 4);
+    sp->setId("z");
+    ListOf_append(lo, sp);
+
+    sp->setId("a");
+    ListOf_append(lo, sp);
+
+    sp->setId("n");
+    ListOf_append(lo, sp);
+
+    lo->sort();
+    SBase_t* child = ListOf_get(lo, 0);
+    fail_unless(child->getId() == "a");
+    child = ListOf_get(lo, 1);
+    fail_unless(child->getId() == "n");
+    child = ListOf_get(lo, 2);
+    fail_unless(child->getId() == "z");
+
```

Comment:
just wondering, what would be the expected outcome when sorting an L2 list of rules?

Reply:
This comment led me to adjust the sorting so it now sorts by, in order:
* the 'id' attribute
* the results of 'getId'
* the name
* the metaid.

So, if the IDs on the rules are set, it sorts by those.  If not, it sorts by the 'variable', since that's returned from a 'getId' call.